modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -322,6 +324,59 @@ CV_EXPORTS_W bool haveImageReader( const String& filename );
  */
 CV_EXPORTS_W bool haveImageWriter( const String& filename );
 
+class CV_EXPORTS ImageCollection {
+    String m_filename;
+    int m_flags;
+    size_t m_size;
+    std::vector<Mat> m_data;
+
+private:
+    explicit ImageCollection(String filename, int flags, size_t size);
+
+public:
+
+    struct Iterator;
+
+    CV_WRAP size_t nimages() const;
+    Mat operator[](int index);
+    CV_WRAP Mat at(int index);
+    CV_WRAP void free(int index);
```

Comment:
does not just release the image at `index` or it also shifts all the subsequent images so that the sequence of images is still contiguous? If the latter, can we use another STL-like name (erase?)